src/ws/hooks/app.ts

```diff
@@ -0,0 +1,180 @@
+/**
+ * Graasp websocket client
+ * React effect hooks to subscribe to real-time updates and mutate query client
+ */
+import { List } from 'immutable';
+import { useEffect } from 'react';
+
+import { UUID, convertJs } from '@graasp/sdk';
+import { AppActionRecord, AppDataRecord, AppSettingRecord } from '@graasp/sdk/frontend';
+
+import { Channel, WebsocketClient } from '../ws-client';
+import { useQueryClient } from '@tanstack/react-query';
+import { buildAppActionsKey, buildAppDataKey, buildAppSettingsKey } from '../../config/keys';
+import { AppActionEvent, AppDataEvent, AppSettingEvent } from '../types';
+import { APP_ACTIONS_TOPIC, APP_DATA_TOPIC, APP_SETTINGS_TOPIC } from '../constants';
+
+export const configureWsAppHooks = (websocketClient: WebsocketClient) => {
+  return {
+    /**
+     * React hook to subscribe to the updates of the app data for
+     * the given item ID.
+     * @param itemId The ID of the item of which to observe updates
+     */
+    useAppDataUpdates: (itemId?: UUID | null) => {
+      const queryClient = useQueryClient();
+      useEffect(() => {
+        if (!itemId || itemId?.length === 0) {
```

Comment:
Not sure if checking the length is a good test. I got confused while reading it, if you want to do proper validation you may want to use [`uuid.validate()`](https://www.npmjs.com/package/uuid#uuidvalidatestr). But I don't think we do that elsewhere in the front-end since they come from the back-end. Let me know if there's another reason do this.

Reply:
I removed the length check to simplify.